nemo/collections/llm/gpt/data/utils.py

```diff
@@ -799,60 +799,93 @@ def _preprocess(
     context_ids = input_ids[:last_ignore_index_pos]
     answer_ids = input_ids[last_ignore_index_pos:]
 
-    return dict(input_ids=input_ids, mask=mask, context_ids=context_ids, answer_ids=answer_ids)
+    return dict(input_ids=input_ids, loss_mask=mask, context_ids=context_ids, answer_ids=answer_ids)
 
 
-def _transform_to_chat_message(source: dict[str, list]):
```

Comment:
please keep this function `_transform_to_chat_message` instead of overwriting it. you can use it as a helper function

Reply:
Is this function used anywhere else?

Reply:
I'd prefer to not have this extra function since it is not called from anywhere outside of this file.

Reply:
Also this function is missing the important injection of the system message at the front of messages.

Reply:
adding system message is good. I meant that separating the `_transform_to_chat_message` from the rest of the apply chat template code would make the code cleaner. Also in the future if we need to transform from ShareGPT/NeMo chat format to HF elsewhere we can reuse the helper function

Reply:
The function is less than 50 lines long at this length the added overhead of another once used private function (prefixed with a _) seems unnecessary. 

But if you are blocking on it, I'm not going to die on this hill.

Reply:
Also the return value of this new function has to be a tuple which also feels unnecessary complex (chat, tools)

Reply:
Restored the separate function.